system/src/Grav/Common/Page/Medium/ImageMedium.php

```diff
@@ -197,28 +192,19 @@ public function cache()
      */
     public function srcset($reset = true)
     {
-        if (empty($this->alternatives) && empty($this->derivatives)) {
+        if (empty($this->alternatives)) {
             if ($reset) {
                 $this->reset();
             }
+
             return '';
         }
 
-        if (!empty($this->derivatives)) {
-          asort($this->derivatives);
-
-          foreach ($this->derivatives as $url => $width) {
-              $srcset[] = $url . ' ' . $width . 'w';
-          }
-
-          $srcset[] = $this->url($reset) . ' ' . $this->get('width') . 'w';
-        }
-        else {
-          $srcset = [ $this->url($reset) . ' ' . $this->get('width') . 'w' ];
```

Comment:
I've done some testing with regards to this. In a case where I have an image with a filename that ends with "@2x", and I call `derivatives()` on that medium, then the fact that the medium is reset by default when calling `srcset()` causes two full-width copies to be generated if `url()` is also called on the medium. This is clearly an edge case, but since I was a bit unsure of what the reset here was for anyway, I thought I should mention it.